File: Level 1/실패율/solution.py
def solution(N, stages):
    challengers = [0] * (N + 2)

    for stage in stages:
        challengers[stage] += 1

    fails = {}
    total = len(stages)

    for i in range(1, N + 1):
        if challengers[i] == 0:
            fails[i] = 0
        else:
            fails[i] = challengers[i] / total
            total -= challengers[i]

    return sorted(fails, key=lambda stage: fails[stage], reverse=True)

File: Level 1/실패율/solution.js
function solution(N, stages) {
  const challengers = new Array(N + 2).fill(0);

  stages.forEach((stage) => {
    challengers[stage]++;
  });

  const fails = {};
  let total = stages.length;

  for (let i = 1; i <= N; i++) {
    if (!challengers[i]) {
      fails[i] = 0;
      continue;
    }

    fails[i] = challengers[i] / total;
    total -= challengers[i];
  }

  return Object.entries(fails)
    .sort((a, b) => b[1] - a[1])
    .map(([stage]) => +stage);
}
